app/controllers/blog_posts_controller.rb

```diff
@@ -4,7 +4,7 @@ class BlogPostsController < ApplicationController
   before_filter :find_blog_post!, :only => :show
 
   def index
-    @blog_posts = BlogPost.all
+    @blog_posts = BlogPost.all.page(params[:page]).per(5)
```

Comment:
We should probably have more than 5 posts on a page (`.per(5)`). Also you should put the number in a constant.